apps/worker/main.py

```
@@ -0,0 +1,171 @@
+
+class ExtractionWorker:
+    """Background worker for processing extraction jobs.
+
+    Polls Redis extraction queue and processes documents through the
+    multi-tier extraction pipeline (Tier A → B → C).
+    """
+
+    def __init__(
+        self,
+        redis_client: redis.Redis,
+        extract_use_case: Optional[Any] = None,
```

Ruff (0.14.1)

Dynamically typed expressions (typing.Any) are disallowed in `extract_use_case`

(ANN401)